transport/interface.go

```diff
@@ -23,15 +23,21 @@ package transport
 import (
 	"time"
 
+	"github.com/opentracing/opentracing-go"
 	"golang.org/x/net/context"
 )
 
 // Request is the fields used to make an RPC.
 type Request struct {
-	Method  string
-	Timeout time.Duration
-	Headers map[string]string
-	Body    []byte
+	Method          string
+	Timeout         time.Duration
+	Headers         map[string]string
+	Baggage         map[string]string
+	RoutingKey      string
+	RoutingDelegate string
+	ShardKey        string
```

Comment:
I don't know if routing key/delegate/shard key belong as top level fields on the request, unless we've made those transport-agnostic concerns in YARPC?